### DISCUSSION TITLE
File upload – additional needs and requests

### DISCUSSION THREAD

**Discussion by mrosvik** [digdir/designsystemet#1861] [Investigate]
[File upload](https://designsystemet.no/no/components/docs/file-upload/overview) was released in Designsystemet in [v.1.19.0](https://github.com/digdir/designsystemet/releases/tag/v1.19.0). 

We’ll keep this thread open for any additional needs or requests that we can consider for the component.

We’ve also received requests for more guidance around file upload, including additional examples and potentially a broader pattern covering some of the challenges that come with file upload flows. This could include things like handling different upload scenarios, communicating status and errors, and accessibility considerations.

Feel free to share relevant use cases, examples, or experiences in this thread. They can help us understand what guidance would be most useful to include alongside the component.

---

## Relevant articles and insight
- https://design-system.service.gov.uk/components/file-upload/
- https://designnotes.blog.gov.uk/2017/02/14/some-design-tips-for-uploading-things/
- https://aksel.nav.no/komponenter/core/fileupload
- https://bootcamp.uxdesign.cc/designing-a-file-upload-component-523e7ca6db4d

### Articles about drag and drop:
- https://www.nngroup.com/articles/drag-drop/
- https://smart-interface-design-patterns.com/articles/drag-and-drop-ux/

**Comment by kimandrefladen** [comment]
Fiskeridirektoratet sin filopplaster ser slik ut:

Opplastet
<img width="816" height="295" alt="Upload" src="https://github.com/user-attachments/assets/dfcc4632-49c8-41a2-9542-47c14db34ee7" />

Opplasting feilet
<img width="816" height="295" alt="Upload failed" src="https://github.com/user-attachments/assets/a3d05321-6c0d-41b3-86d7-612dd2fb0df5" />

Slett opplastet fil
<img width="816" height="295" alt="Delete uploaded file" src="https://github.com/user-attachments/assets/be3e299e-291e-432e-aad6-58e6bca06c42" />

**Comment by Camulos** [comment]
Improvement to the file-upload component?
How to show uploaded content, guidance for some much used patterns could be part of the documentation:
For example:

- list of files, with filename, size and contextual tools like "remove file"
- single upload with renaming based on input/category
- preview of image or document
- security or virusscan checks?
